apps/managedidentity/servicefabric.go

```diff
@@ -0,0 +1,129 @@
+package managedidentity
+
+import (
+	"context"
+	"crypto/sha1"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+)
+
+func miVerifyCertificate(cert *x509.Certificate, verifiedChain [][]*x509.Certificate, customPool *x509.CertPool) error {
```

Comment:
I feel that this logic is too complex to go in without proper E2E testing. Can we move it post-GA ? I.e. in GA we reach parity with Azure SDK and we take this as an enhancement.

Reply:
I agree with you on this,
we can shift this to post GA